gcloud/storage/bucket.py

```diff
@@ -738,7 +725,9 @@ def make_public(self, recursive=False, future=False):
         self.acl.save()
 
         if future:
-            doa = self.get_default_object_acl()
+            doa = self.default_object_acl
+            if not doa.loaded:
+                doa.reload()
```

Comment:
Nope. `make_public` needs work / deletion.

An even scarier line of code is immediately below:

```python
        if recursive:
            for blob in self:
```

which could retrieve millions of blobs on a large bucket.